tests/libnet/namespace.go

```diff
@@ -0,0 +1,99 @@
+package libnet
+
+import (
+	"context"
+	"encoding/json"
+
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	"kubevirt.io/client-go/kubecli"
+)
+
+func AddLabelToNamespace(namespace, key, value string) {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	ns, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	old, err := json.Marshal(ns)
+	Expect(err).ToNot(HaveOccurred())
+	new := ns.DeepCopy()
+	if new.Labels == nil {
+		new.Labels = map[string]string{}
+	}
+	new.Labels[key] = value
+
+	newJson, err := json.Marshal(new)
+	Expect(err).ToNot(HaveOccurred())
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	_, err = virtCli.CoreV1().Namespaces().Patch(context.Background(), ns.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func RemoveLabelFromNamespace(namespace, key string) {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	ns, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	old, err := json.Marshal(ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	new := ns.DeepCopy()
+	if new.Labels == nil {
+		// There are no labels, nothing to do
+		return
+	}
+	delete(new.Labels, key)
+
+	newJson, err := json.Marshal(new)
+	Expect(err).ToNot(HaveOccurred())
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	_, err = virtCli.CoreV1().Namespaces().Patch(context.Background(), ns.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func RemoveAllLabelsFromNamespace(namespace string) {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	ns, err := virtCli.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	old, err := json.Marshal(ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	new := ns.DeepCopy()
+	if new.Labels == nil {
+		// There are no labels, nothing to do
+		return
+	}
+	new.Labels = map[string]string{}
+
+	newJson, err := json.Marshal(new)
+	Expect(err).ToNot(HaveOccurred())
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, ns)
+	Expect(err).ToNot(HaveOccurred())
+
+	_, err = virtCli.CoreV1().Namespaces().Patch(context.Background(), ns.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func PanicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
I don't think you need to copy this function here; just receive the kubevirtClient as a parameter to all the functions that need it.

It might be a bit cumbersome, but, it is what we're doing everywhere else (unless I'm mistaken).

Reply:
Updated as you suggested